handlers/oauthhandler/oauthhandler.go

```diff
@@ -0,0 +1,182 @@
+
+// Init does nothing
+func (dh *OauthHandler) Init(opts ...string) error {
+	return nil
+}
+
+// GetName returns the name of the handler
+func (dh *OauthHandler) Name() string {
+	return "oAuth"
+}
+
+// Info returns the handler options and required auth steps.
+func (dh *OauthHandler) Info() *types.Message {
+	return &types.Message{
+		Title:    "oAuth handler",
+		AuthType: "auth",
+		SignType: types.AllSignatures,
+		AuthSteps: []*types.AuthField{
+			{Title: "GetAuthUrl", Type: "text"},
+			{Title: "VerifyElection", Type: "text"},
+		},
+	}
+}
+
+// Indexer takes a unique user identifier and returns the list of processIDs where
+// the user is elegible for participation. This is a helper function that might not
+// be implemented (depends on the handler use case).
+func (ih *OauthHandler) Indexer(userID types.HexBytes) []types.Election {
+	return nil
+}
+
+// Auth is the handler for the dummy handler
+func (dh *OauthHandler) Auth(r *http.Request,
+	c *types.Message, pid types.HexBytes, signType string, step int,
+) types.AuthResponse {
+	if signType != types.SignatureTypeBlind {
+		return types.AuthResponse{Response: []string{"incorrect signature type, only blind supported"}}
+	}
+
+	providers, err := Init(pid.String())
+	if err != nil {
+		return types.AuthResponse{Response: []string{"failed to initialize providers"}}
+	}
+
+	switch step {
+	case 0:
+		if len(c.AuthData) != 2 {
+			return types.AuthResponse{Response: []string{"missing auth data"}}
+		}
+		service := c.AuthData[0]
+		redirectURL := c.AuthData[1]
+		atoken := uuid.New()
+
+		provider, ok := providers[service]
+		if !ok {
+			return types.AuthResponse{Response: []string{"Provider not found."}}
+		}
+
+		// Get the Service Auth URL from the electionID and requested service
+		authURL := provider.GetAuthURL(redirectURL)
+
+		return types.AuthResponse{
+			Success:   true,
+			AuthToken: &atoken,
+			Response:  []string{authURL},
+		}
+	case 1:
+		// Convert the provided "code" to an oAuth Token
+		if len(c.AuthData) != 3 {
+			return types.AuthResponse{Response: []string{"auth token not provided or missing auth data"}}
+		}
+		service := c.AuthData[0]
+		oAuthCode := c.AuthData[1]
+		redirectURL := c.AuthData[2]
+		provider, ok := providers[service]
+		if !ok {
+			log.Errorf("Provider not found. %s", service)
+			return types.AuthResponse{Response: []string{"Provider not found."}}
+		}
+
+		oAuthToken, err := provider.GetOAuthToken(oAuthCode, redirectURL)
+		if err != nil {
+			log.Errorf("error obtaining the oAuthToken: %s", err)
+			return types.AuthResponse{Response: []string{"error obtaining the oAuthToken"}}
+		}
+
+		// Get the profile
+		profileRaw, err := provider.GetOAuthProfile(oAuthToken)
+		if err != nil {
+			log.Errorf("error obtaining the profile: %s", err)
+			return types.AuthResponse{Response: []string{"error obtaining the profile"}}
+		}
+
+		var profile map[string]interface{}
+		jerr := json.Unmarshal([]byte(profileRaw), &profile)
+		if err != nil {
```

Comment:
I think you are not checking the right error here, you declare `jerr` and you are checking err. This kind of errors could be avoided using the pattern that I previously mentioned, `if := err ...  ; err != nil `